src/shogun/features/CombinedDotFeatures.cpp

```diff
@@ -129,67 +135,53 @@ float64_t CCombinedDotFeatures::dense_dot(int32_t vec_idx1, const float64_t* vec
 
 void CCombinedDotFeatures::dense_dot_range(float64_t* output, int32_t start, int32_t stop, float64_t* alphas, float64_t* vec, int32_t dim, float64_t b) const
```

Comment:
`dense_dot_range` and `dense_dot_range_subset` ignored the feature_weights, while other dot functions used it, so I had to change them before writing tests